lib/shortcuts/stream-lambda.js

```diff
@@ -60,6 +62,10 @@ class StreamLambda extends Lambda {
       }
     };
 
+    if (FilterCriteria) {
+      this.Resources[`${this.LogicalName}EventSource`].Properties.FilterCriteria = FilterCriteria;
+    }
```

Comment:
Lets add some test coverage for this

Reply:
Can we also add some validation on the value passed through if it is indeed an object? Even better if we can do some validation on the properties.

Reply:
Added some validation